### PR TITLE
Collections api

### PR DESCRIPTION
**Pull request by ryanrdoherty**
Kinda massive PR covering collections API, two other new metadata vars (entity many-to-one and impute_zero), factoring of distribution config into separate objects shared by vars and collections, and reworking DB constants into a hierarchy.

### CODE REVIEW

**Comment by ryanrdoherty** [comment]
@Foxcapades and @dmgaldi Please focus on the following files in your review:

Class encapsulating DB schema:
https://github.com/VEuPathDB/EdaSubsettingService/pull/63/files#diff-5dec80f580c67a45167859124aa61102a5b8285601ae362703976f805246d568

Class that loads the new collections metadata:
https://github.com/VEuPathDB/EdaSubsettingService/pull/63/files#diff-b0e9d20e2e9fea0b70e76c85aa71482127d83ab280c4b1b8368000489dad6084

Parent collection class (performs some important checks and finishes populating fields based on member vars)
https://github.com/VEuPathDB/EdaSubsettingService/pull/63/files#diff-2b79a9618e6d76fe6384c9ffb9e0cb2a914a7f89db3b185a1d77f56a70c1a4ab

File: src/main/java/org/veupathdb/service/eda/ss/model/db/DB.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+import org.veupathdb.service.eda.ss.model.Entity;
+
+import static org.veupathdb.service.eda.ss.model.db.DB.ColumnCollection.getAll;
+import static org.veupathdb.service.eda.ss.model.db.DB.ColumnCollection.getAllAsStream;
+
+public interface DB {
+
+  interface Tables {
+
+    interface Study {
```
Comment (dmgaldi) [inline src/main/java/org/veupathdb/service/eda/ss/model/db/DB.java:18]: Just a thought, could a reflectionless solution be achieved by using enums and the associated `values()` method for these inner classes? There might be reasons not to go this route that I'm not considering.

File: src/main/java/org/veupathdb/service/eda/ss/model/db/EntityFactory.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    // entityID -> list of child entities
+    Map<String, List<Entity>> childrenMap = new HashMap<>();
+
+    Entity rootEntity = new SQLRunner(_dataSource, sql, "Get entity tree").executeQuery(rs -> {
+      Entity root = null;
+      while (rs.next()) {
+        Entity entity = createEntityFromResultSet(rs);
+        String parentId = rs.getString(DB.Tables.EntityTypeGraph.Columns.ENTITY_PARENT_ID_COL_NAME);
+        if (parentId == null) {
+          if (root != null) throw new RuntimeException("In Study " + studyId + " found more than one root entity");
```
Comment (dmgaldi) [inline src/main/java/org/veupathdb/service/eda/ss/model/db/EntityFactory.java:48]: Maybe worth including the roots here

File: src/main/java/org/veupathdb/service/eda/ss/model/varcollection/VarCollection.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            "data type and shape as its parent collection " + _properties.id);
+      }
+      VariableWithValues valueVar = (VariableWithValues)var.get();
+      valueVars.add(valueVar);
+      if (valueVar.getVocabulary() != null && !valueVar.getVocabulary().isEmpty()) {
+        derivedVocabulary.addAll(valueVar.getVocabulary());
+      }
+      else {
+        // do not declare a vocabular unless all member vars have a vocabulary
+        //LOG.warn("Member variable " + varId + " of collection " + _properties.id + " does not have a vocabulary.");
```
Comment (dmgaldi) [inline src/main/java/org/veupathdb/service/eda/ss/model/varcollection/VarCollection.java:95]: Not sure if intentionally left commented, or intended to remove/uncomment

File: src/main/java/org/veupathdb/service/eda/ss/model/db/DB.java (hunk 1 above)
Comment (ryanrdoherty) [reply src/main/java/org/veupathdb/service/eda/ss/model/db/DB.java:18]: It's a good idea.  I'm not sure of pros/cons right now.  Going to leave as-is for now and we can readdress later.

File: src/main/java/org/veupathdb/service/eda/ss/model/db/EntityFactory.java (hunk 2 above)
Comment (ryanrdoherty) [reply src/main/java/org/veupathdb/service/eda/ss/model/db/EntityFactory.java]: Done.

File: src/main/java/org/veupathdb/service/eda/ss/model/varcollection/VarCollection.java (hunk 3 above)
Comment (ryanrdoherty) [reply src/main/java/org/veupathdb/service/eda/ss/model/varcollection/VarCollection.java:95]: When I put it in, there were a ton of outputs.  I'm not exactly sure of the long-term shape of the data, i.e. whether most of these values will have vocabs, but I'm actually thinking maybe they mostly won't, so will remove this line.

**Review by Foxcapades — APPROVED** [review]

File: src/main/java/org/veupathdb/service/eda/ss/model/varcollection/VarCollection.java (hunk 3 above)
Comment (ryanrdoherty) [reply src/main/java/org/veupathdb/service/eda/ss/model/varcollection/VarCollection.java]: I added some more docs and put this log in a different place to make clear what's going on.

**Pull request merged**